src/Admin/AbstractAdmin.php

```diff
@@ -2848,8 +2850,16 @@ public function id($model)
         return $this->getNormalizedIdentifier($model);
     }
 
+    /**
+     * NEXT_MAJOR: Remove this method.
+     */
     public function setValidator($validator)
     {
+        @trigger_error(sprintf(
```

Comment:
This method is automatically called by the CompilerPass, so I think we can't trigger an error inside (or everybody will have a deprecation everytime). @sonata-project/contributors ?